### ISSUE TITLE
Using `initialize="true"` leads to inconsistent decision for skipping zero sample

### ISSUE THREAD

**Issue by davidscn** [precice/precice#1998]
**Describe your setup**

preCICE Version: 3.1.0

**Describe the problem**
My testcase (doesn't depend on the number of participants) requires to use `<exchange... intialize="true"/>`. I run in parallel and not all participants are part of the coupling interface. In our current implementation, we call `mapData` with `skipZero=true` and use a logical `&&` to connect it to data being zero or not

https://github.com/precice/precice/blob/4c3f798231b131583106b882b36af175b6b99538/src/precice/impl/DataContext.cpp#L129

The other participant nicely provides initial data and the condition `stample.sample.values.isZero()` evaluates to `false` for all participants, who are part of the coupling interface. However, for all participants not part of the interface,  `stample.sample.values.size() == 0`, and therefore `stample.sample.values.isZero()` evaluates to `true`.

This inconsistencies leads to decisions on individual ranks to enter `mapData` or skip it. However, we have mappings relying on collective MPI operations, such as the Eigen RBF or (didn't check, but should apply) our PETSc based mappings, such that we run into a deadlock for these cases.

The decision to skip the zero sample or not should be consistent across ranks. Not all mapping variants rely on collective MPI operations, but from a logical perspective, it would make sense to synchronize the decision (in my case, the logging on the primary rank was also wrong, as it didn't hold any interface fraction).

**Step To Reproduce**
1. Setup a participant and run it at least on two ranks. One of the ranks should not be part of the interface. Configure `initialize="true"` and initialize non-zero data.
2. Configure one of our global RBF mapping variants.
3. Run into a deadlock.

**Comment by davidscn** [comment]
Regarding a solution. The cleanest way would probably be to use `utils::IntraComm::l2norm(stample.sample.values)` instead of the `isZero()` function to perform consistent operations across all ranks. That's of course not the only option: Given that we don't really need the `l2norm` here, we could use `MPI_Allreduce` with `MPI_LOR` to synchronize the boolean evaluated on each rank.

A more hacky solution (which also wouldn't work in case data on a rank is actually zero, whereas it is non-zero on other ranks) would be to not skipZero on ranks, which are not at the interface.
